server/clusters/ism/ismPlugin.ts

```diff
@@ -15,485 +14,8 @@ export default function ismPlugin(Client: any, config: any, components: any) {
-          type: "string",
-          required: true,
-        },
-      },
-    },
-    needBody: true,
-    method: "PUT",
-  });
-
-  ism.putTransform = ca({
-    url: {
-      fmt: `${API.TRANSFORM_BASE}/<%=transformId%>`,
-      req: {
-        transformId: {
-          type: "string",
-          required: true,
-        },
-      },
-    },
-    method: "PUT",
-  });
-
-  ism.previewTransform = ca({
-    url: {
-      fmt: `${API.TRANSFORM_BASE}/_preview`,
-    },
-    needBody: true,
-    method: "POST",
-  });
-
-  ism.getChannels = ca({
-    url: {
-      fmt: `${API.CHANNELS_BASE}`,
-    },
-    method: "GET",
-  });
-
-  ism.getChannel = ca({
-    url: {
-      fmt: `${API.NOTIFICATION_CONFIGS_BASE}/<%=id%>`,
-      req: {
-        id: {
-          type: "string",
-          required: true,
-        },
-      },
-    },
-    method: "GET",
-  });
-
-  ism.getSMPolicy = ca({
-    url: {
-      fmt: `${API.SM_POLICY_BASE}/<%=id%>`,
-      req: {
-        id: {
-          type: "string",
-          required: true,
-        },
-      },
-    },
-    method: "GET",
-  });
-
-  ism.getSMPolicies = ca({
-    url: {
-      fmt: `${API.SM_POLICY_BASE}`,
-    },
-    method: "GET",
-  });
-
-  ism.createSMPolicy = ca({
-    url: {
-      fmt: `${API.SM_POLICY_BASE}/<%=policyId%>?refresh=wait_for`,
-      req: {
-        policyId: {
-          type: "string",
-          required: true,
-        },
-      },
-    },
-    needBody: true,
-    method: "POST",
-  });
-
-  ism.updateSMPolicy = ca({
-    url: {
-      fmt: `${API.SM_POLICY_BASE}/<%=policyId%>?if_seq_no=<%=ifSeqNo%>&if_primary_term=<%=ifPrimaryTerm%>&refresh=wait_for`,
-      req: {
-        policyId: {
-          type: "string",
-          required: true,
-        },
-        ifSeqNo: {
-          type: "string",
-          required: true,
-        },
-        ifPrimaryTerm: {
-          type: "string",
-          required: true,
-        },
-      },
-    },
-    needBody: true,
-    method: "PUT",
-  });
-
-  ism.deleteSMPolicy = ca({
-    url: {
-      fmt: `${API.SM_POLICY_BASE}/<%=policyId%>?refresh=wait_for`,
-      req: {
-        policyId: {
-          type: "string",
-          required: true,
-        },
-      },
-    },
-    method: "DELETE",
-  });
-
-  ism.explainSnapshotPolicy = ca({
-    url: {
-      fmt: `${API.SM_POLICY_BASE}/<%=id%>/_explain`,
-      req: {
-        id: {
-          type: "string",
-          required: true,
-        },
-      },
-    },
-    method: "GET",
-  });
-
-  ism.startSnapshotPolicy = ca({
-    url: {
-      fmt: `${API.SM_POLICY_BASE}/<%=id%>/_start`,
-      req: {
-        id: {
-          type: "string",
-          required: true,
-        },
-      },
-    },
-    method: "POST",
-  });
-
-  ism.stopSnapshotPolicy = ca({
-    url: {
-      fmt: `${API.SM_POLICY_BASE}/<%=id%>/_stop`,
-      req: {
-        id: {
-          type: "string",
-          required: true,
-        },
-      },
-    },
-    method: "POST",
+  extendClient({
```

Comment:
I saw the logic is moved to clusters/extend_client.ts, there is no much logic left in the ismPlugin.ts file, what's the benefit of break it to two files?

Reply:
We can reuse the logic of constructing an endpoint `ism` by moving all the `ism.xxx = ca(xxx)` code into a separate file.